src/dune_rules/lib.ml

```diff
@@ -431,9 +427,9 @@ let wrapped t =
       assert false (* will always be specified in dune package *)
     | Some (This x) -> Some x)
 
-let equal l1 l2 = Ordering.is_eq (compare l1 l2)
+let equal = ( == )
```

Comment:
I don't quite follow. What's wrong with comparing the path and name of a library? Not sure where the ids mentioned in the PR description are coming from. If from the name, why can't we use strings for library names instead?

Reply:
In watch mode, a library with the same and path and name can change as dune files are edited. When such a change occurs, the `Lib.t` changes but the path and the name stay the same. Therefore, it's not correct to treat such libraries as equal.

Reply:
I see, thanks. 

Shouldn't we just implement a proper equality check instead of relying on the physical equality that might some day become a wrong thing to do? It's not like comparing all these fields is going to be expensive. (Perhaps some future profiling will prove me wrong, but I'd rather not optimise things unnecessarily.)

Reply:
Comparing these fields correctly would require forcing a lot of lazy values unfortunately.

Even if a better equality would become available one day, the current code should stay correct. So the worst case that could happen is extra re-computation.

Reply:
OK, I see. It seems worth leaving a comment saying this.

Reply:
Done.